src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const splittedValue = sourceString.split(';');
+
+  for (let i = 0; i < splittedValue.length; i++) {
+    const entry = splittedValue[i].split(':');
+
+    if (entry.length === 2) {
+      const property = entry[0].trim();
+      const value = entry[1].trim();
+
+      result[property] = value;
+    }
+  }
```

Comment:
Instead of using a for loop, consider using array iteration methods such as 'map' and 'filter' to transform the string into an object. This would be more in line with functional programming practices and improve code readability.